File: Class Demo/VB.Net/OOPsConcept/OOPsConcept/Programmer.vb
Public Class Programmer
    Inherits Emp
    Dim extraHrs As Integer
    Dim costPerHrs As Double
    Public Sub New()
        MyBase.New()  'MyBase keyword is used to refer to base class
        extraHrs = 0
        costPerHrs = 0
    End Sub

    Public Sub New(ByVal e As Integer, ByVal nm As String,
                   ByVal bs As Double, ByVal eh As Integer,
                   ByVal costPerHrs As Double)
        MyBase.New(e, nm, bs)
        extraHrs = eh
        Me.costPerHrs = costPerHrs
        'Me is used in place of this keyword
    End Sub
    'Function hiding is known as shadowing 
    Public Overrides Sub Display()
        MyBase.Display()
        Console.WriteLine("Extra Hrs=" + extraHrs.ToString())
        Console.WriteLine("Cost Per Hrs=" + costPerHrs.ToString())
    End Sub

    Public Overrides Function CalSal() As Double
        Return basic + extraHrs * costPerHrs
    End Function


End Class

File: C# Assignment/assignment solutions/ass1/2/Programmer.cs
using System;

namespace _2
{
    class Programmer:Emp
    {
        double costPerHrs;
        int extraHrs;
        
        public Programmer()
        {
            costPerHrs = extraHrs=0;
        }

        public Programmer(string nm,int d,int m,int y,int i,int s,int c,int e):base(nm,d,m,y,i,s)
        {
            costPerHrs = c;
            extraHrs = e;
        }

        public void calsal()
        {
            Console.WriteLine("Total salary : "+(base.calsal() + extraHrs * costPerHrs));
        }

        public void display()
        {
            base.display();
            calsal();
        }

        public void accept()
        {
            base.accept();
            Console.WriteLine("Enter the extra hours");
            extraHrs=int.Parse(Console.ReadLine());
            Console.WriteLine("Enter the cost per hours");
            costPerHrs = int.Parse(Console.ReadLine());
        }

    }
}
